setup.sh

```diff
@@ -0,0 +1,382 @@
+#!/bin/bash
+set -e
+
+# list of colors
+BOLD=$(tput bold)
```

Comment:
Before defining colors, you should check if the terminal supports them, and then make sure the defined colors are compatible with the terminal: http://unix.stackexchange.com/questions/9957/how-to-check-if-bash-can-print-colors